sdks/python/apache_beam/io/source_test_utils.py

```diff
@@ -0,0 +1,572 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from collections import namedtuple
+import logging
+
+from multiprocessing.pool import ThreadPool
+from apache_beam.io import iobase
+
+
+class ExpectedSplitOutcome(object):
+  MUST_SUCCEED_AND_BE_CONSISTENT = 1
+  MUST_FAIL = 2
+  MUST_BE_CONSISTENT_IF_SUCCEEDS = 3
+
+
+SplitAtFractionResult = namedtuple(
+    'SplitAtFractionResult', 'num_primary_items num_residual_items')
+
+SplitFractionStatistics = namedtuple(
+    'SplitFractionStatistics',
+    'successful_fractions non_trivial_fractions')
+
+
+class SourceTestUtils(object):
+  """Helper functions and test harnesses for source implementations.
+
+  This module contains helper functions and test harnesses for checking
```

Comment:
Yeah, this sounds better.

Moved class methods to be top level functions and moved the doc comment to the module level.